### PR TITLE
feat(config-cli): New config-cli package includes dump basemaps screenshots command line tool

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/config-cli/src/cli/screenshot/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env node
+import { BaseCommandLine } from '@basemaps/shared/build/cli/base.js';
+import 'source-map-support/register.js';
+import { CommandScreenShot } from './screenshot.js';
+
+export class ScreenshotCommandLine extends BaseCommandLine {
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/index.ts:6]: lets call this `BasemapsConfig` as we will add more sub tools to it like the config import

`bmc` as the script name? or `basemaps-config` ?

File: packages/config-cli/src/cli/screenshot/screenshot.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+      await page.goto(url);
+
+      try {
+        if (host.startsWith('dev')) {
+          await page.waitForSelector('div#map-loaded', { state: 'attached' });
+          await page.waitForTimeout(1000);
+          await page.waitForLoadState('networkidle');
+        } else {
+          throw new Error('Not supported on production yet');
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/screenshot.ts:135]: this is now in production we can take this logic out.

File: packages/config-cli/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Basemaps Config CLI
+
+This package is to control the configuration in the LINZ basemaps product.
+
+## Usage -- Screenshots
+
+Dump the screenshots from basemaps production
+
+```bash
+./bmc.js screenshot
```
Comment (blacha) [inline packages/config-cli/README.md:10]: I dont see a `./bmc.js` do these commands work?

File: packages/config-cli/bmc.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env node
```
Comment (blacha) [inline packages/config-cli/bmc.js:1]: this file should be referenced by the bin field in the package.json, it also wont be published.

Its best to put this into `./bin/bmc.mjs` (being specific is super helpful for these)

then add `bin/` to the files exported and add a `bin` section to map it to `basemaps-screenshot` ?

File: packages/config-cli/src/cli/screenshot/screenshot.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    });
+
+    this.tiles = this.defineStringParameter({
+      argumentName: 'TILES',
+      parameterLongName: '--tiles',
+      description: 'JSON file path for the test tiles',
+      defaultValue: './test-tiles/default.test.tiles.json',
+    });
+
+    this.verbose = this.defineFlagParameter({
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/screenshot.ts:56]: verbose is handled by being a part of `BaseCommandLine` 

`./bmc --verbose screenshot`

File: packages/config-cli/src/cli/screenshot/screenshot.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    } finally {
+      await chrome.close();
+    }
+  }
+
+  async takeScreenshots(chrome: Browser, logger: LogType): Promise<void> {
+    const host = this.host.value ?? this.host.defaultValue;
+    const tag = this.tag.value ?? this.tag.defaultValue;
+    const tiles = this.tiles.value ?? this.tiles.defaultValue;
+    if (host == null || tag == null || tiles == null)
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/screenshot.ts:82]: can this ever happen?

File: packages/config-cli/README.md (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/config-cli/README.md:10]: Yes, that's how does this work for me without install the package.

Comment (blacha) [reply packages/config-cli/README.md:10]: ahh was meaning in the `package.json` this file would not have be published. I expanded on it in my next comment

File: packages/config-cli/src/cli/screenshot/screenshot.ts (hunk 5 above)
Comment (Wentao-Kuang) [reply packages/config-cli/src/cli/screenshot/screenshot.ts]: All fixed. Can you give another look?

File: packages/config-cli/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# Basemaps Config CLI
+
+This package is to control the configuration in the LINZ basemaps product.
+
+## Usage -- Screenshots
+
+Dump the screenshots from basemaps production
+
+```bash
+./bin/bmc.js screenshot
```
Comment (blacha) [inline packages/config-cli/README.md:10]: NIT: if its described as `bin : { "bmc": ... }` the actual command will be `bmc` once you npm install it

File: packages/config-cli/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+```
+
+Dump the screenshots from different host and tag
+
+```bash
+.bin//bmc.js screenshot --host HOST --tag PR-TAG
```
Comment (blacha) [inline packages/config-cli/README.md:16]: NIT: `//`

File: packages/config-cli/package.json [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    "bmc": "./bin/bmc.js"
+  },
+  "scripts": {
+    "test": "ospec --globs 'build/**/*.test.js'"
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "files": [
+    "build/"
```
Comment (blacha) [inline packages/config-cli/package.json:31]: missing `bin/`

File: packages/config-cli/src/cli/screenshot/index.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env node
+import { BaseCommandLine } from '@basemaps/shared/build/cli/base.js';
+import 'source-map-support/register.js';
+import { CommandScreenShot } from './screenshot.js';
+
+export class BasemapsConfig extends BaseCommandLine {
+  constructor() {
+    super({
+      toolFilename: 'bin',
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/index.ts:9]: `bmc`

File: packages/config-cli/src/cli/screenshot/index.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env node
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/index.ts:1]: this file should never be executed directly.

File: packages/config-cli/src/cli/screenshot/screenshot.ts [hunk 12]
```diff
@@ -0,0 +1,142 @@
+import { Config, fsa, LogConfig, LogType } from '@basemaps/shared';
+import { mkdir } from 'fs/promises';
+import { Browser, chromium } from 'playwright';
+import { CommandLineAction, CommandLineStringParameter } from '@rushstack/ts-command-line';
+import { z } from 'zod';
+
+enum TileMatrixIdentifier {
+  NZTM = 'NZTM2000Quad',
```
Comment (blacha) [inline packages/config-cli/src/cli/screenshot/screenshot.ts:8]: Do we not have this as a enum somewhere else?

We should be specific here `Nztm2000Quad` so we know we are talking about nztm2000quad tms and not the other one.

**Review by blacha — APPROVED** [review]

**Pull request merged**